test/extended/default_forcePull_test.go

```diff
@@ -0,0 +1,262 @@
+	s2iDockBldr       = "docker.io/openshift/ruby-20-centos7"
+	custBldr          = "docker.io/openshift/origin-custom-docker-builder"
+	s2iStartingHexID  string
+	custStartingHexID string
+)
+
+/*
+If docker.io is not responding to requests in a timely manner, this test suite will be adversely affected.
+
+If you suspect such a situation, attempt pulling some openshift images other than ruby-20-centos7 or origin-custom-docker-builder
+while this test is running and compare results.  Restarting your docker daemon, assuming you can ping docker.io quickly, could
+be a quick fix.
+*/
+
+func corruptImage(corruptee, corruptor, strategy string) {
+	By(fmt.Sprintf("\n%s****BEFORE EACH FORCE PULL*****:  Calling docker tag to corrupt %s builder image %s by tagging %s", time.Now().Format(time.RFC850), strategy, corruptee, corruptor))
+
+	cerr := exutil.TagImage(corruptee, corruptor)
+
+	By(fmt.Sprintf("\n%s****BEFORE EACH FORCE PULL*****:  Tagging %s to %s complete with err %v", time.Now().Format(time.RFC850), corruptor, corruptee, cerr))
+	Expect(cerr).NotTo(HaveOccurred())
+}
+
+func resetImage() {
+	// reset both images just to be sure
+	By(fmt.Sprintf("\n%s****FORCE PULL*****:  Calling docker tag to reset images", time.Now().Format(time.RFC850)))
+
+	serr := exutil.TagImage(s2iDockBldr, s2iStartingHexID)
+
+	By(fmt.Sprintf("\n%s****FORCE PULL*****:  Reset for %s to %s complete with err %v", time.Now().Format(time.RFC850), s2iStartingHexID, s2iDockBldr, serr))
+	Expect(serr).NotTo(HaveOccurred())
+
+	cerr := exutil.TagImage(custBldr, custStartingHexID)
+
+	By(fmt.Sprintf("\n%s****FORCE PULL*****:  Reset for %s to %s complete with err %v", time.Now().Format(time.RFC850), custStartingHexID, custBldr, cerr))
+	Expect(cerr).NotTo(HaveOccurred())
+
+}
+
+func verifyImagesSame(strategy string) {
+	s2iComp := s2iDockBldr + ":latest"
+	custComp := custBldr + ":latest"
+
+	comps := []string{s2iComp, custComp}
+	retIDs, gerr := exutil.GetImageIDForTags(comps)
+
+	Expect(gerr).NotTo(HaveOccurred())
+	By(fmt.Sprintf("\n%s******FORCE PULL TEST %s false ******:  compare image - %s, %s, %s, %s", time.Now().Format(time.RFC850), strategy, s2iComp, custComp, retIDs[0], retIDs[1]))
+	Expect(len(retIDs[0]) > 0).Should(BeTrue())
+	Expect(len(retIDs[1]) > 0).Should(BeTrue())
+	Expect(retIDs[0]).To(Equal(retIDs[1]))
+}
+
+func verifyImagesDifferent(strategy string) {
+	s2iComp := s2iDockBldr + ":latest"
+	custComp := custBldr + ":latest"
+
+	comps := []string{s2iComp, custComp}
+	retIDs, gerr := exutil.GetImageIDForTags(comps)
+
+	Expect(gerr).NotTo(HaveOccurred())
+	By(fmt.Sprintf("\n%s******FORCE PULL TEST %s true ******:  compare image - %s, %s, %s, %s", time.Now().Format(time.RFC850), strategy, s2iComp, custComp, retIDs[0], retIDs[1]))
+	Expect(len(retIDs[0]) > 0).Should(BeTrue())
+	Expect(len(retIDs[1]) > 0).Should(BeTrue())
+	Expect(retIDs[0] != retIDs[1]).Should(BeTrue())
+}
+
+func waitForBuild(context string, oc *exutil.CLI) {
```

Comment:
can you pass the buildName as parameter for this function? i think it will make it more usable for other test cases :-)